### PR TITLE
Seo indexing

### PR DESCRIPTION
**Pull request by craig-mulligan**
So basic steps to get google to index a javascript site. 

1. Google notices that your page is rendered using Javascript. (sees the hashbang #! in your url)
2. Google requests your pages with a modified url. (replaces #! with ?_escaped_fragment_=)
3. You return the prerendered HTML to the crawler. 

So we could host our own pre-rendering HTML and do the custom routing, but I found a service (prerender.io) that has a phantom browser that generates the static HTML and caches it. You can have 250 unique urls(pages) for free.

Before merging we'd need create a resin account and add our token as an envar. 

@emirotin @shaunmulligan

### CODE REVIEW

**Comment by dfunckt** [comment]
If we get to actually render static pages, why not also serve them directly?

**Comment by craig-mulligan** [comment]
@dfunckt we could but it would require a fair amount of work generate them and replace our current logic in `main.js`. This is meant as a quick fix as we are considering finding static site generator to replace our current system.

File: static/scripts/main.js [hunk 1]
```diff
@@ -245,7 +249,7 @@ angular
         var url = location.href;
         url = url.replace(/#\w.+/, '')
```
Comment (emirotin) [inline static/scripts/main.js]: this seems to also needs updating to `#!`, no?

File: static/scripts/main.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ var UnpinOffset = 400;
 window.onload = function () {
     // fix first page load anchor issue
     var url = window.location.hash;
-    var divid = url.split('#');
+    var divid = url.split('#!');
```
Comment (emirotin) [inline static/scripts/main.js]: this looks wrong
this code splits the _has_ itself to find a hash-in-hash that identifies an element on the page (like heading) to scroll to it

**Comment by emirotin** [comment]
First, a question: how would you set the env var? The site is hosted on github pages, right?

Second, as far as I know google should actually do normal crawling for JS pages and escaped_fragment shouldn't be required:
http://searchengineland.com/tested-googlebot-crawls-javascript-heres-learned-220157
http://www.infoq.com/news/2014/05/google-renders-css-js-hints

Third, this PR changes the URLs scheme, so it will make all the old URLs (in our UI, cross-linked in docs, in blog posts, in user's bookmarks, everywhere) broken, right?

**Comment by craig-mulligan** [comment]
@emirotin I assumed it was hosted on heroku, because of the `Procfile`. It would break urls but if the crawler can do this without using _escaped_fragment_ URLs and rendering the html itself then I can't see why our docs could not be indexed?

**Comment by emirotin** [comment]
Procfile makes sense, it's the gh-pages branch that makes me confused.
Maybe the key is `#!` in the URLs (check the most recent google link).
We should use webmaster tools and check how google sees our pages. Some of the linked articles should give insights into this process, I think.
Breaking URLs is super-bad :( If me manage to organize redirects (with angular-router's $otherwise clause) it's OK then, but first let's run the investigation.

**Pull request closed without merging**